pkg/handlers/config.go

```diff
@@ -190,6 +195,15 @@ func (h *Handler) UpdateAppConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	validationErrors := kotsadmvalidator.ValidateConfigSpec(kotsv1beta1.ConfigSpec{Groups: updateAppConfigRequest.ConfigGroups})
+	if len(validationErrors) > 0 {
+		updateAppConfigResponse.Error = "failed to validate config values"
+		updateAppConfigResponse.ValidationErrors = validationErrors
+		logger.Error(errors.Wrap(err, updateAppConfigResponse.Error))
```

Comment:
Do we also want to consider logging the validation errors so they end up in support bundles?  Might not be needed since we are going to surface client-side, but could be interesting to see.

Reply:
Do we also want to consider logging the validation errors so they end up in support bundles?  Might not be needed since we are going to surface client-side, but could be interesting to see.

Reply:
We could log according the customer response

Reply:
```
{"level":"error","ts":"2023-04-12T23:11:43Z","msg":"failed to validate config values, validation errors: [{Name:example_settings Title:My Example Configs ItemErrors:[{Name:password Type:password Value:{Type:0 BoolVal:false StrVal:} ValidationErrors:[{ValidationErrorMessage:Value does not match regex RegexValidator:0xc00046ede0}]} {Name:email_text Type:text Value:{Type:0 BoolVal:false StrVal:} ValidationErrors:[{ValidationErrorMessage:Value does not match regex RegexValidator:0xc00046ee00}]} {Name:cve_text_area Type:textarea Value:{Type:0 BoolVal:false StrVal:} ValidationErrors:[{ValidationErrorMessage:Value does not match regex RegexValidator:0xc00046ee20}]} {Name:jwt_file Type:file Value:{Type:0 BoolVal:false StrVal:} ValidationErrors:[{ValidationErrorMessage:Value does not match regex RegexValidator:0xc00046ee40}]}]}]"}
```

logging with a standard formatter `%+v` should be good for debugging